Mathlib/Data/MLList/Basic.lean

```diff
@@ -0,0 +1,26 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Std.Data.MLList.Basic
+
+/-!
+# Basic operations on `MLList`
+
+These functions can be upstreamed to Std when convenient.
+-/
+
+set_option autoImplicit true
+
+namespace MLList
+
+/-- Construct a singleton monadic lazy list from a single monadic value. -/
+def singletonM [Monad m] (x : m α) : MLList m α :=
+  .squash fun _ => do return .cons (← x) .nil
```

Comment:
Should this be a `MonadLift` instance?

Reply:
I think it is useful to have the function explicitly named?